### PR TITLE
ansible: clean instance, rebuild contracts and reenable services

### PR DESCRIPTION
**Pull request by danzvara**
Relates to #83

### CODE REVIEW

File: ansible/playbooks/create-meter.yaml [hunk 1]
```diff
@@ -44,27 +44,44 @@
...
+             && echo "  unlock = ['`cat /energy/client_addr`']" >> /energy/parity/res/config.toml
+             && echo "  password = [\"/energy/client_pass\"]" >> /energy/parity/res/config.toml
+    - name: install truffle
+      shell: npm install -g truffle
+    - name: build contracts
+      shell: rm -rf /energy/truffle/build && cd /energy/truffle && truffle compile && truffle build
     - name: reload systemd
       shell: systemctl daemon-reload
+    - name: Disable ParityEnergy service if running
+      shell: systemctl disable parity-energy && systemctl stop parity-energy
```
Comment (fangyi-zhou) [inline ansible/playbooks/create-meter.yaml]: stop then disable?

File: ansible/playbooks/create-meter.yaml [hunk 2]
```diff
@@ -44,27 +44,44 @@
...
+      ignore_errors: yes
```
Comment (danzvara) [inline ansible/playbooks/create-meter.yaml]: Ignoring errors because systemctl has stupid return codes

File: ansible/playbooks/create-meter.yaml [hunk 3]
```diff
@@ -44,27 +44,44 @@
...
     - name: Enable ParityEnergy service
       shell: systemctl enable /energy/parity/parity-energy.service
     - name: Start ParityEnergy service
       shell: systemctl start parity-energy
     - name: Build Meter server
       shell: cd /energy/client/meter && npm install
+    - name: Disable Meter service if running
+      shell: systemctl disable meter-server && systemctl stop meter-server
```
Comment (fangyi-zhou) [inline ansible/playbooks/create-meter.yaml]: ditto

File: ansible/playbooks/create-meter.yaml (hunk 2 above)
Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: ok

File: ansible/playbooks/create-meter.yaml (hunk 1 above)
Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: Also, if disable fails, it cannot be stopped.

Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: Doesn't matter, disable just kills symlinks

File: ansible/playbooks/create-meter.yaml (hunk 3 above)
Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: ditto

File: ansible/playbooks/create-meter.yaml [hunk 4]
```diff
@@ -44,27 +44,44 @@
       shell: pwgen 10 1 > /energy/client_pass
       args:
         creates: /energy/client_pass
+    - name: clean old parity accounts
+      shell: rm -rf /energy/parity/datadir/keys/*
     - name: create new account
       shell: parity --chain /energy/parity/res/genesis.json --keys-path /energy/parity/datadir//keys account new --password /energy/client_pass > /energy/client_addr
     - name: get money to the account
       shell: curl -X POST -d `cat /energy/client_addr` http://146.169.47.73:6000/money
       ignore_errors: yes
```
Comment (fangyi-zhou) [inline ansible/playbooks/create-meter.yaml]: Run `sed -i '' '/\[account\]*$/,$d' res/config.toml` to truncate old account

Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: account -> config

File: ansible/playbooks/create-meter.yaml (hunk 1 above)
Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: ok

File: ansible/playbooks/create-meter.yaml (hunk 4 above)
Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: (y)

File: ansible/playbooks/create-meter.yaml [hunk 5]
```diff
@@ -44,27 +44,46 @@
...
+             && echo "  unlock = ['`cat /energy/client_addr`']" >> /energy/parity/res/config.toml
+             && echo "  password = [\"/energy/client_pass\"]" >> /energy/parity/res/config.toml
+    - name: install truffle
+      shell: npm install -g truffle
+    - name: build contracts
+      shell: rm -rf /energy/truffle/build && cd /energy/truffle && truffle compile && truffle build
     - name: reload systemd
       shell: systemctl daemon-reload
+    - name: Disable ParityEnergy service if running
+      shell: systemctl stop parity-energy && systemctl disable parity-energy
```
Comment (fangyi-zhou) [inline ansible/playbooks/create-meter.yaml]: What will happen if stop fails?

Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: Bad things. Also stop won't really fail unless the service is already disabled

Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: ok

File: ansible/playbooks/create-meter.yaml (hunk 4 above)
Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: This doesn't work becasue GNU sed != BSD sed

Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: Oops

Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: Resolved

Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: Do we need to fix `deploy-vm.sh` as well?

Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: Don't think so, we're not using it on VM's and on OSX it's fine

Comment (fangyi-zhou) [reply ansible/playbooks/create-meter.yaml]: @vincepandolfo uses Linux

Comment (danzvara) [reply ansible/playbooks/create-meter.yaml]: Eh, local fix? :D Anyway that doesn't belong to this PR

**Comment by fangyi-zhou** [comment]
What are you talking about, we use blockchain because we don't trust each other

**Pull request merged**